### PR TITLE
feat: add support for GraalVM build

### LINKED ISSUE #36: Add support for GraalVM builds in CI
### Feature Description

Previous builds passed with GraalVM, but a recent CI-related change has broken them. This is the issue for restoring CI builds against GraalVM.

### Sample code

_No response_

### Alternatives Considered

_No response_

### Additional Details

_No response_

### PR DESCRIPTION
**Pull request by ttosta-google**
Fixes #36

As discussed the GraalVM job will run on periodic (schedule) event as it takes from 7 to 10min to finish ([example](https://github.com/GoogleCloudPlatform/alloydb-java-connector/actions/runs/6711868339/job/18240203950?pr=228))

### CODE REVIEW

File: alloydb-jdbc-connector/pom.xml [hunk 1]
```diff
@@ -167,5 +168,32 @@
...
+      classpath, so adding these drivers back in for the native profile is a
+      workaround.
+    -->
+    <profile>
+      <id>native</id>
+      <dependencies>
+        <dependency>
+          <groupId>org.postgresql</groupId>
+          <artifactId>postgresql</artifactId>
+          <version>${postgresql.version}</version>
```
Comment (enocom) [inline alloydb-jdbc-connector/pom.xml:194]: Is this necessary for our final artifact or just for test?

File: alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/nativeimage/AlloyDBFeature.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import org.graalvm.nativeimage.hosted.RuntimeReflection;
+
+/**
+ * Registers GraalVM configuration for the AlloyDB libraries.
+ *
+ * <p>This class is only used when this library is used in <a
+ * href="https://www.graalvm.org/22.0/reference-manual/native-image/">GraalVM native image</a>
+ * compilation.
+ */
+public class AlloyDBFeature implements Feature {
```
Comment (enocom) [inline alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/nativeimage/AlloyDBFeature.java:29]: Does this class need to be public?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -111,80 +111,89 @@ jobs:
...
-          #        # only run flakybot on periodic (schedule) and continuous (push) events
-          #        if: ${{ (github.event_name == 'schedule' || github.event_name == 'push') && runner.os == 'Linux' && always() }}
-          #        run: |
-          #          curl https://github.com/googleapis/repo-automation-bots/releases/download/flakybot-1.1.0/flakybot -o flakybot -s -L
-          #          chmod +x ./flakybot
-          #          ./flakybot --repo ${{github.repository}} --commit_hash ${{github.sha}} --build_url https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}
-          #
+  graalvm17:
+    # run job on periodic (schedule) and continuous (push) events
+    if: "${{ (github.event_name == 'schedule' || github.event_name == 'push') && always() }}"
```
Comment (enocom) [inline .github/workflows/ci.yaml:116]: Dumb question: what is push? Pushes to PRs and merges to main?

Comment (ttosta-google) [reply .github/workflows/ci.yaml:116]: A `push` event is created when we merge a PR to the `main` branch as we configured:

```
on:
  push:
    branches:
    - main
```

[official documentation](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-only-when-a-push-to-specific-branches-occurs)

Comment (jackwotherspoon) [reply .github/workflows/ci.yaml:116]: @ttosta-google  Why do we only want this on `push` and `schedule` events and not on pull requests? I also don't think you want the `always()` here...

**Review by jackwotherspoon — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (ttosta-google) [reply .github/workflows/ci.yaml]: Updated to run on `schedule` as agreed. Removed `always()`

File: alloydb-jdbc-connector/pom.xml (hunk 1 above)
Comment (ttosta-google) [reply alloydb-jdbc-connector/pom.xml:194]: Updated.

**Review by enocom — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -111,80 +111,91 @@ jobs:
...
+        ALLOYDB_INSTANCE_NAME: '${{ steps.secrets.outputs.ALLOYDB_INSTANCE_NAME }}'
+        ALLOYDB_INSTANCE_IP: '${{ steps.secrets.outputs.ALLOYDB_INSTANCE_IP }}'
+        ALLOYDB_IMPERSONATED_USER: '${{ steps.secrets.outputs.ALLOYDB_IMPERSONATED_USER }}'
+        JOB_TYPE: graalvm17
+      run: .kokoro/build.sh
+      shell: bash
+
+    - name: FlakyBot
+      # only run flakybot on periodic (schedule) and continuous (push) events
+      if: ${{ (github.event_name == 'schedule' || github.event_name == 'push') && runner.os == 'Linux' && always() }}
```
Comment (jackwotherspoon) [inline .github/workflows/ci.yaml:194]: If the parent level job is only being run on `schedule` then this child check is unnecessary and can be removed
```suggestion
      # run flakybot even on failures
      if: always()
```

**Review by jackwotherspoon — APPROVED** [review]

**Review by jackwotherspoon — CHANGES_REQUESTED** [review]
nit: remove child logic that is unnecessary

File: .github/workflows/ci.yaml (hunk 4 above)
Comment (ttosta-google) [reply .github/workflows/ci.yaml]: Updated. Thanks!

**Review by jackwotherspoon — APPROVED** [review]

**Pull request merged**